src/puppetlabs/puppetdb/http/server.clj

```diff
@@ -30,9 +42,30 @@
     (format "The %s API has been retired; please use v4" version)
     404)))
 
-(def routes
-  (app
-   ["v4" &] {:any v4-app}
+(defn query-app
```

Comment:
how is this going to evolve when new endpoints are added for v5? stick the endpoints in here and put the v4 rejection logic in the relevent http namespace?